### ISSUE TITLE
[Clarification] Normalized to Pelvis?

### ISSUE THREAD

**Issue by bryanbocao** [Xinyu-Yi/TransPose#73]
Dear Authors,

Thank you for sharing the code. I am trying to understand the ```rotation to the root``` part:

https://github.com/Xinyu-Yi/TransPose/blob/2b31fd1bf1534c6f9c973fa091c0c8cbcccc7310/utils.py#L10-L16

It looks like ```acc``` and ```ori``` sensing data in the global coordinate frame are normalized/rotated to the root, which is the last one of the ```6``` IMU devices, while the last one's index of ```0``` in ```ji_mask``` shows that it is the ```pelvis``` according to
https://github.com/Xinyu-Yi/TransPose/blob/2b31fd1bf1534c6f9c973fa091c0c8cbcccc7310/preprocess.py#L34-L35

So the code normalizes the global sensor data (acceleration and orientation) to the pelvis's coordinate system. Is this understanding correct?

Thank you in advance.

**Comment by Xinyu-Yi** [comment]
Exactly. The last IMU is placed on the pelvis(root) joint. All the other 5 IMUs are transformed to the last IMU's frame. Because we have a T-pose calibration before the transformation, the last IMU's frame is exactly the root joint's frame.

**Comment by bryanbocao** [comment]
@Xinyu-Yi Thank you for helping me understand the code!

Now if in my code I have a different masks where the pelvis(root) is placed at the beginning:
```
 vi_mask = torch.tensor([3021, ...]) 
 ji_mask = torch.tensor([0, ...])
```
To make sensors aligned with the root's coordinate system, it should be like:
```
acc = torch.cat((glb_acc[:, 1:] - glb_acc[:, :1], glb_acc[:, :1]), dim=1).bmm(glb_ori[:, 0]) / acc_scale 
ori = torch.cat((glb_ori[:, :1].transpose(2, 3).matmul(glb_ori[:, 1:]), glb_ori[:, :1]), dim=1) 
```

**Comment by Xinyu-Yi** [comment]
Yes. But you also need to check if other places are using the order. Also, your acc and ori are now still in root-last order, which is different from glb_acc and glb_ori. Make sure these orders are correctly used in the rest codes.
